nullaway/src/test/resources/com/uber/nullaway/testdata/ReadBeforeInitNegativeCases.java

```diff
@@ -159,4 +164,114 @@ private void initF() {
       this.baz = new Object();
     }
   }
+
+  static class SingleInitializer {
+
+    Object f;
+
+    SingleInitializer() {
+      f = new Object();
+    }
+
+    @Initializer
+    public void init() {
+      f.toString();
+    }
+  }
+
+  static class SingleInitializer2 {
+
+    Object f;
+    Object g;
+
+    SingleInitializer2() {
+      f = new Object();
+      g = new Object();
+    }
+
+    SingleInitializer2(boolean b) {
+      if (b) {
+        f = new Object();
+      } else {
+        f = "hi";
+      }
+      g = new Object();
+    }
+
+    @Initializer
+    public void init() {
+      g.toString();
+      f.toString();
+    }
+  }
+
+  static class SingleStaticInitializer {
+
+    static Object f;
+    static Object g;
+
+    @Initializer
+    static void init() {
+      g.toString();
+      f.toString();
+    }
+
+    static {
+      f = new Object();
+      g = new Object();
+    }
+  }
+
+  static class CompareToNullInInit {
+
+    Object f;
+
+    @Initializer
+    void init() {
+      if (f == null) {
+        f = new Object();
+      }
+    }
+  }
+
+  static class CompareToNullInInit2 {
+
+    Object f;
+
+    @Initializer
+    void init() {
+      if (null == f) {
+        f = new Object();
+      }
+    }
+  }
+
+  static class CompareToNullInInit3 {
+
+    Object f;
+
+    @Initializer
+    void init() {
+      if (!(f != null)) {
+        f = new Object();
+      }
+    }
+  }
+
+  static class CastToNonNullTest {
+
+    Object castF;
+    Object castG;
+
+    @Initializer
+    void init(@Nullable Object o) {
+      if (o != null) {
+        castF = castToNonNull(castF);
```

Comment:
Cool.

Reply:
Yeah I saw this in our code base; very cool local suppression technique!